src/main/java/org/embulk/util/guess/GuesstimatedType.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 The Embulk project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.embulk.util.guess;
+
+import java.util.Objects;
+
+/**
+ * Represents (Embulk's) data type estimated by guess.
+ *
+ * <p>It reimplements {@code SchemaGuess.GuessedType} in {@code /embulk/guess/schema_guess.rb}.
+ *
+ * @see <a href="https://github.com/embulk/embulk/blob/v0.10.19/embulk-core/src/main/ruby/embulk/guess/schema_guess.rb">schema_guess.rb</a>
+ */
+public class GuesstimatedType implements Comparable<GuesstimatedType> {
+    private GuesstimatedType(final String string, final String formatOrTimeValue) {
+        this.string = string;
+        this.formatOrTimeValue = formatOrTimeValue;
+    }
+
+    private GuesstimatedType(final String string) {
+        this(string, null);
+    }
+
+    static GuesstimatedType timestamp(final String formatOrTimeValue) {
+        return new GuesstimatedType("timestamp", formatOrTimeValue);
+    }
+
+    /**
+     * Checks if the type is {@code TIMESTAMP}.
+     *
+     * @return {@code true} if the type is {@code TIMESTAMP}
+     */
+    public boolean isTimestamp() {
+        return this.string.equals("timestamp");
```

Comment:
better `"timestamp".equals(this.string)`

Reply:
I actually sometimes do that, but it's "not always" better. Nice reading: https://en.wikipedia.org/wiki/Yoda_conditions#Criticism